application-diagram-ui/src/main/resources/Diagram/DiagramEditSheet.xml

```diff
@@ -421,10 +421,182 @@ define('diagram-store', ['jquery', 'xwiki-meta', 'xwiki-utils', 'diagram-utils',
+  var addTagSpecificStyle = function(element, s, w, h, htmlConverter, str, alt) {
+    switch (element.tagName.toLowerCase()){
+      case 'h1':
+        alt.setAttribute('font-size', Math.round(s.fontSize + 0.6 * s.fontSize) + 'px');
+        alt.setAttribute('font-weight', 'bold');
+        break;
+      case 'h2':
+        alt.setAttribute('font-size', Math.round(s.fontSize + 0.3 * s.fontSize) + 'px');
+        alt.setAttribute('font-weight', 'bold');
+        break;
+      case 'h3':
+        alt.setAttribute('font-size', Math.round(s.fontSize + 0.15 * s.fontSize) + 'px');
+        alt.setAttribute('font-weight', 'bold');
+        break;
+      case 'h4':
+        alt.setAttribute('font-weight', 'bold');
+        break;
+      case 'h5':
+        alt.setAttribute('font-size', Math.round(s.fontSize - 0.15 * s.fontSize) + 'px');
+        alt.setAttribute('font-weight', 'bold');
+        break;
+      case 'h6':
+        alt.setAttribute('font-size', Math.round(s.fontSize - 0.3 * s.fontSize) + 'px');
+        alt.setAttribute('font-weight', 'bold');
+        break;
+      default:
+        return alt;
+    }
+    return alt;
+  };
+
+  var getStrWidth = function(str, s) {
+    return $('&lt;span&gt;&lt;/span&gt;')
+      .css({display: 'none', whiteSpace: 'nowrap'})
+      .appendTo($('body'))
+      .text(str)
+      .width();
+  };
+
+  // Creates a row inside a text element that should be wrapped. The coordinates of the first row are already saved.
+  var createRow = function(w, h, rowNb) {
+    var row = document.createElementNS('http://www.w3.org/2000/svg', 'tspan');
+    row.setAttribute('x', w);
+    row.setAttribute('y', h);
+    if (rowNb &gt; 0) {
+      takenCoordinates.push({'w': w, 'h': h});
+    }
+    return row;
+  };
+
+  var wrapText = function(alt, s, w, h, content, contentWidth, maxWidth) {
+    var words = content.split(/[\s+]/);
+    var wordWidth, rowWidth = 0, rowNb = 0;
+    var row = createRow(w, h, rowNb);
+    alt.append(row);
+    words.each(function(word) {
+      word += ' ';
+      wordWidth = getStrWidth(word, s);
+      if (rowWidth + wordWidth &gt; maxWidth) {
+        h += 15;
+        row = createRow(w, h, ++rowNb);
+        row.textContent = word;
+        alt.append(row);
+        rowWidth = wordWidth;
+      } else {
+        row.textContent += word;
+        rowWidth += wordWidth;
+      }
+    });
+  };
+
+  var takenCoordinates = [];
+  // For the current block of text take into consideration if those coordinates are already in use.
+  // This is used when we are computing manually the container of children elements.
+  var maybeChangeHeight = function(w, h, isChildElement) {
+    if (!isChildElement) {
+      return h;
+    }
+    var thisCoordinates = {'w': w, 'h': h};
+    takenCoordinates.each(function(coordinates) {
+      if (coordinates['w'] == thisCoordinates['w'] &amp;&amp; coordinates['h'] == thisCoordinates['h']) {
+        h += 15;
```

Comment:
I see you've used 15 above also, and there's a 30 below. I assume this is the line height. Would be good to define a local variable in this module to keep the line height.